cpp/src/IceUtil/StringConverter.cpp

```diff
@@ -2,17 +2,19 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-#if defined(_MSC_VER) && (_MSVC_LANG >= 201703L)
-    // TODO codecvt was deprecated in C++17 and cause build failures with VC++ compiler
-    // we should replace this code with MultiByteToWideChar() and WideCharToMultiByte()
-#   define  _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING
-#endif
-
 #include <IceUtil/StringConverter.h>
 #include <IceUtil/MutexPtrLock.h>
 #include <IceUtil/Mutex.h>
 #include <IceUtil/StringUtil.h>
 
+// TODO codecvt was deprecated in C++17 and cause build failures in C++17 mode
+#if (ICE_CPLUSPLUS >= 201703L)
+#   include <IceUtil/DisableWarnings.h>
+#elif defined(_MSC_VER) && (_MSVC_LANG >= 201703L)
```

Comment:
Is this now dead code?

I assume we set ICE_CPLUSPLUS when defined(_MSC_VER) && (_MSVC_LANG >= 201703L)

Reply:
right we can switch the condition the other way around

Reply:
I reworked the check